daemon/analytics.c

```diff
@@ -0,0 +1,881 @@
+{
+    if (chdir(dir) == -1)
+        fatal("Cannot change directory to '%s'", dir);
+
+    DIR *d = opendir(dir);
+    if (!d)
+        fatal("Cannot examine the contents of directory '%s'", dir);
+    closedir(d);
+
+    return dir;
+}
+
+/*
+ * This is called after the rrdinit
+ * These values will be sent on the START event
+ */
+void set_late_global_environment()
+{
+    analytics_set_data(&analytics_data.netdata_config_stream_enabled, default_rrdpush_enabled ? "true" : "false");
+    analytics_set_data_str(&analytics_data.netdata_config_memory_mode, (char *)rrd_memory_mode_name(default_rrd_memory_mode));
+    analytics_set_data(&analytics_data.netdata_config_exporting_enabled, appconfig_get_boolean(&exporting_config, CONFIG_SECTION_EXPORTING, "enabled", 1) ? "true" : "false");
+
+#ifdef ENABLE_DBENGINE
+    {
+        char b[16];
+        snprintfz(b, 15, "%d", default_rrdeng_page_cache_mb);
+        analytics_set_data(&analytics_data.netdata_config_page_cache_size, b);
+
+        snprintfz(b, 15, "%d", default_multidb_disk_quota_mb);
+        analytics_set_data(&analytics_data.netdata_config_multidb_disk_quota, b);
+    }
+#endif
+
+#ifdef ENABLE_HTTPS
+    analytics_set_data(&analytics_data.netdata_config_https_enabled, "true");
+#else
+    analytics_set_data(&analytics_data.netdata_config_https_enabled, "false");
+#endif
+
+    if (web_server_mode == WEB_SERVER_MODE_NONE)
+        analytics_set_data(&analytics_data.netdata_config_web_enabled, "false");
+    else
+        analytics_set_data(&analytics_data.netdata_config_web_enabled, "true");
+
+    analytics_set_data_str(&analytics_data.netdata_config_release_channel, (char *)get_release_channel());
+
+    {
+        BUFFER *bi = buffer_create(1000);
+        analytics_build_info(bi);
+        analytics_set_data_str(&analytics_data.netdata_buildinfo, (char *)buffer_tostring(bi));
+        buffer_free(bi);
+    }
+
+    /* set what we have, to send the START event */
+    analytics_setenv_data();
+}
+
+static void get_system_timezone(void)
+{
+    // avoid flood calls to stat(/etc/localtime)
+    // http://stackoverflow.com/questions/4554271/how-to-avoid-excessive-stat-etc-localtime-calls-in-strftime-on-linux
+    const char *tz = getenv("TZ");
+    if (!tz || !*tz)
+        setenv("TZ", config_get(CONFIG_SECTION_GLOBAL, "TZ environment variable", ":/etc/localtime"), 0);
+
+    char buffer[FILENAME_MAX + 1] = "";
+    const char *timezone = NULL;
+    ssize_t ret;
+
+    // use the TZ variable
+    if (tz && *tz && *tz != ':') {
+        timezone = tz;
+        info("TIMEZONE: using TZ variable '%s'", timezone);
+    }
+
+    // use the contents of /etc/timezone
+    if (!timezone && !read_file("/etc/timezone", buffer, FILENAME_MAX)) {
+        char *c = buffer;
+        while (*c) {
+            if (unlikely(*c == '\n'))
+                *c = '\0';
+            c++;
+        }
+        timezone = buffer;
+        info("TIMEZONE: using the contents of /etc/timezone: '%s'", timezone);
```

Comment:
This appears now on startup. Do we really need to show it?

Reply:
@stelfrag netdata uses different possible methods to select the timezone, if this is removed from the startup message, I think it would be good to insert this information inside our `error.log` for we have conditions to debug the origin of the timezone.

Reply:
I guess it will be helpful for figuring out timezone issues to have it logged